config.proto

```diff
@@ -127,3 +130,17 @@ message JavaAgent {
     // filter_jar_paths is the list of path to filter jars, separated by `,`
     repeated google.protobuf.StringValue filter_jar_paths = 1;
 }
+
+// CustomDataCaptureEndpoint represents a custom data capture rule for an endpoint that matches
+// the given url pattern and host header
+message CustomDataCaptureEndpoint {
+
+    // data_capture describes the data being captured by instrumentation for this endpoint
+    DataCapture data_capture = 1;
```

Comment:
It's nice to describe this in terms of agent's `DataCapture` config instead of just having a single flag like `capture_payload`. This has implication on what we need to propagate: https://github.com/hypertrace/specification/pull/7. It will not just be a single 0-1 flag to propagate in the header. Can we propagate the DataCapture definition? Otherwise we can propagate a matched rule ID, which downstream modules can lookup its `DataCapture` config definition of. Maybe the latter is nicer.

Reply:
I would avoid to mix capture and propagation here because that is overkill. Locally services know what to capture and I don't think upstream should override that.